modules/detectron/upsample_nearest_op.h

```diff
@@ -35,8 +35,54 @@ class UpsampleNearestOp final : public Operator<Context> {
   USE_OPERATOR_CONTEXT_FUNCTIONS;
 
   bool RunOnDevice() override {
-    // No CPU implementation for now
-    CAFFE_NOT_IMPLEMENTED;
+    auto translate_idx = [](int ii, int d1, int d2, int d3, int scale_factor) {
+      int x, y, z, w;
```

Comment:
I'm not sure if this function will be called for many times. If so, consider to use FixedDivisor to make div and mod faster.
https://github.com/pytorch/pytorch/blob/master/caffe2/utils/fixed_divisor.h